### ISSUE TITLE
rspconfig openbmc not working with hierarchy

### ISSUE THREAD

**Issue by mattaezell** [xcat2/xcat-core#4805]
From the management node, these commands don't work:
```
[root@mgmt1.peak ~]# lsxcatd -v
Version 2.13.10 (git commit 4fe63ecd9f836f3cf447118d461e1c10c69b1652, built Fri Jan 26 00:09:41 EST 2018)
[root@mgmt1.peak ~]# rspconfig h41n01 powersupplyredundancy
h41n01: Error: [400] Failed to convert the types
[root@mgmt1.peak ~]# rspconfig h41n01 autoreboot
h41n01: Error: [400] Failed to convert the types
```
Directly from the service node, they do:
```
[root@mgmt1.peak ~]# ssh sn1
Last login: Wed Feb  7 16:04:32 2018 from mgmt1.peak.olcf.ornl.gov
[root@sn1 ~]# lsxcatd -v
Version 2.13.10 (git commit 4fe63ecd9f836f3cf447118d461e1c10c69b1652, built Fri Jan 26 00:09:41 EST 2018)
[root@sn1 ~]# rspconfig h41n01 powersupplyredundancy
h41n01: BMC PowerSupplyRedundancy : Disabled
[root@sn1 ~]# rspconfig h41n01 autoreboot           
h41n01: BMC AutoReboot: 0
```

**Comment by mattaezell** [comment]
@gurevichmark FYI

**Comment by gurevichmark** [comment]
@mattaezell Can you run these commands with xcatdebugmode turned on ?

```chdef -t site xcatdebugmode=1``` to turn on
```chdef -t site xcatdebugmode``` to turn off

**Comment by mattaezell** [comment]
```
[root@mgmt1.peak ~]# chdef -t site xcatdebugmode=1
1 object definitions have been created or modified.
[root@mgmt1.peak ~]# rspconfig h41n01 powersupplyredundancy
Wed Feb  7 17:50:02 2018 OpenBMC: [openbmc_debug]
h41n01: Error: [400] Failed to convert the types
```

journal on mgmt1:
```
Feb 07 17:50:02 mgmt1.peak.olcf.ornl.gov xcat[7722]: DEBUG xcatd: connection from root@localhost
Feb 07 17:50:02 mgmt1.peak.olcf.ornl.gov xcat[7722]: DEBUG xcatd: open new process : xcatd SSL: rspconfig to h41n01 for root@localhost
Feb 07 17:50:02 mgmt1.peak.olcf.ornl.gov xcat[7724]: DEBUG xcatd: dispatch request 'rspconfig h41n01 powersupplyredundancy' to plugin 'openbmc'
Feb 07 17:50:02 mgmt1.peak.olcf.ornl.gov xcat[7722]: xCAT: Allowing rspconfig to h41n01 powersupplyredundancy for root from localhost
Feb 07 17:50:02 mgmt1.peak.olcf.ornl.gov xcat[7724]: DEBUG xcatd: handle request 'rspconfig' by plugin 'openbmc''s preprocess_request
Feb 07 17:50:02 mgmt1.peak.olcf.ornl.gov xcat[7724]: DEBUG [openbmc_debug] OpenBMC
Feb 07 17:50:02 mgmt1.peak.olcf.ornl.gov xcat[7726]: DEBUG dispatch hierarchical sub-command rspconfig to sn1:3001
Feb 07 17:50:06 mgmt1.peak.olcf.ornl.gov xcat[7722]: DEBUG xcatd: close connection with root@localhost
```

Journal on sn1:
```
Feb 07 17:50:02 sn1 xcat[14886]: DEBUG xcatd: connection from mgmt1.peak.olcf.ornl.gov@mgmt1
Feb 07 17:50:02 sn1 xcat[14886]: DEBUG xcatd: open new process : xcatd SSL: rspconfig for mgmt1.peak.olcf.ornl.gov@mgmt1
Feb 07 17:50:02 sn1 xcat[14887]: DEBUG xcatd: dispatch request 'rspconfig powersupplyredundancy' to plugin 'openbmc'
Feb 07 17:50:02 sn1 xcat[14886]: xCAT: Allowing rspconfig to h41n01 powersupplyredundancy for root from mgmt1
Feb 07 17:50:02 sn1 xcat[14887]: DEBUG xcatd: handle request 'rspconfig' by plugin 'openbmc''s process_request
Feb 07 17:50:06 sn1 xcat[14886]: DEBUG xcatd: close connection with mgmt1.peak.olcf.ornl.gov@mgmt1
```

**Comment by mattaezell** [comment]
FYI, rflash does still work correctly

**Comment by hu-weihua** [comment]
@tingtli , could you help to verify this fix? thanks

**Comment by tingtli** [comment]
I think this is fixed by 2.14 4/3 build.
Here is some information
```
[root@boston02 ~]# lsxcatd -v
Version 2.14 (git commit f7867c9e516f28843b498e69399c6029adf52bca, built Tue Apr  3 06:15:43 EDT 2018)
[root@boston02 ~]# lsdef -l mid08tor03cn10 -i servicenode
Object name: mid08tor03cn10
    servicenode=sn02
[root@boston02 ~]# rspconfig mid08tor03cn10 powersupplyredundancy
mid08tor03cn10: BMC PowerSupplyRedundancy: Enabled
[root@boston02 ~]# rspconfig mid08tor03cn10 autoreboot
mid08tor03cn10: BMC AutoReboot: 1
```

**Comment by tingtli** [comment]
@mattaezell  Please take a look. thanks!
